### DISCUSSION TITLE
References?

### DISCUSSION THREAD

**Discussion by copumpkin** [voneiden/cq-cam#9] [General]
I saw https://cq-cam.readthedocs.io/en/latest/algorithms.html wasn't written yet, but I was wondering if maybe your readme could list some reading materials for folks who want to get up to speed in this space and possibly help. CAM toolpath generation seems to mostly be proprietary research and code and it's remarkably hard to find any documentation on techniques, so I'm curious if you've been figuring it out on your own or if there are books or other resources you could recommend.

**Comment by voneiden** [comment]
Indeed, I was surprised to find out the same thing when I started working on this - there's not much reference material available. I thought I have a file somewhere with relevant research paper links but can't seem to find it right now. However I do recall two papers that were interesting reads.

* [Optimal CNC Tool Path Generation for Zigzag Pocket Milling (2001) by T. El-Midany](https://www.researchgate.net/publication/283089115_Optimal_CNC_Tool_Path_Generation_for_Zigzag_Pocket_Milling) served as a stepping stone for me to implement a simple zigzag algorithm. 

* [CNC PROGRAMMING OF POCKET MACHINING (2017) by M. Bošnjaković](https://www.researchgate.net/publication/317441548_CNC_PROGRAMMING_OF_POCKET_MACHINING) some general considerations

The algorithms I have here are fairly simple - for 2D operations we're mostly just manipulating Wires that are being offset and clipped as needed, and then converted into G-code. 

I'm not entirely sure yet what my next step will be with this project. I kinda want to sit down and review the code this summer and put some effort into cleaning up the messy parts. My priority earlier this year during active development phase was just to get a solid proof of concept together. A strong foundation would seem fairly important as I expect this project will keep growing over the years.

**Comment by voneiden** [comment]
I've converted this as a discussion. I did finally some experimenting with #1 and I have a pretty good feeling about it. It might be possible to significantly simplify the current implementation.

**Comment by copumpkin** [comment]
Awesome, thanks for the references and help! My interests are primarily in the 4th axis space which I realize is pretty advanced but I have some ideas I’d like to experiment with but the amount of foundation “spatial manipulation” work I need to do to start writing the toolpaths I’m thinking of is huge so I was looking for some good basis to build off.

I’ll eagerly watch your progress in this repo, thanks!

**Comment by giannissc** [comment]
- [Learning CNC Strategies and Tip](https://cnc-programming-tips.blogspot.com/?m=0)
- [Support CNC](https://mhcc.pressbooks.pub/supportcnc/)
- [CNC Cookbook Gcode Basics](https://www.cnccookbook.com/g-code-basics-program-format-structure-blocks/)

**Reply by voneiden** [reply]
[On the Computational Geometry of Pocket Machining](https://link.springer.com/book/10.1007/3-540-54103-9) might be an interesting read at some point.
